lib/berlim/company_accounts/employee.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Berlim.CompanyAccounts.Employee do
+  @moduledoc false
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Berlim.CompanyAccounts.{Company, Sector}
+
+  schema "employees" do
+    field :active, :boolean, default: true
+    field :internal_id, :string
+    field :name, :string
+    belongs_to :company, Company
+    belongs_to :sector, Sector
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(employee, attrs) do
+    employee
+    |> cast(attrs, [:name, :internal_id, :active, :company_id, :sector_id])
+    |> validate_required([:name])
+    |> cast_or_constraint_assoc(:company)
+    |> cast_or_constraint_assoc(:sector)
+  end
+
+  defp cast_or_constraint_assoc(changeset, name) do
```

Comment:
qual intenção desse método? n entendi

Reply:
É pra validar que a associação ``belongs_to`` não seja nula, caso eu tente fazer um insert, sem passar o ``company_id`` ou ``sector_id``, utilizando só o ``assoc_constraint``, lança uma exceção do Postgre: 

> (Postgrex.Error) ERROR 23502 (not_null_violation) null value in column "company_id" violates not-null constraint

E esse método verifica, se houver alguma "change" no atributo passado como parâmetro, aplique o ``assoc_constraint``, senão, aplique o ``cast_assoc`` com o ``required:true``, para retornar a mensagem de que o campo é obrigatório, ao invés de lançar o erro do Postgre.

Reply:
duvida: não seria o fluxo normal estourar no Postgres e o elixir trata isso? não é assim que acontece com validações de uniq_constraint?

Reply:
Não, o ``unique_constraint`` tbm não valida se o field que tá sendo passado como parâmetro está nulo ou não, creio que quem faz essa validação, seja só o ``validate_required``.

Tomando como exemplo o model de Táxi, se remover o ``:email`` do ``validate_required`` e permanecer no ``unique_constraint``, ao tentar inserir um táxi sem passar o email, vai lançar a mesma exceção acima.

Reply:
Tendi

Reply:
@danpassoss tava testando com @thiagogsr aqui e o que precisamos fazer nesse caso é adicionar `validate_required([:name, :company_id])` e `foreign_key_constraint(:company_id)` e `foreign_key_constraint(:sector_id)` algo como:
```
def changeset(employee, attrs) do
    employee
    |> cast(attrs, [:name, :internal_id, :active, :company_id, :sector_id])
    |> validate_required([:name, :company_id])
    |> foreign_key_constraint(:company_id)
    |> foreign_key_constraint(:sector_id)
  end
```

Reply:
o retorno quando for enviado o sector_id e o sector não existir é esse: 
```
{:error,
 #Ecto.Changeset<
   action: :insert,
   changes: %{active: true, company_id: 1, name: "danilo", sector_id: 2},
   errors: [
     sector_id: {"does not exist",
      [constraint: :foreign, constraint_name: "employees_sector_id_fkey"]}
   ],
   data: #Berlim.CompanyAccounts.Employee<>,
   valid?: false
 >}
```

Reply:
Ok, eu havia estudado essa possibilidade de usar o ``validate_required``, porém a própria documentação não recomenda, por isso busquei outra forma.

Reply:
qual doc dan?

Reply:
do ``validate_required``

Reply:
> Do not use this function to validate associations are required, instead pass the :required option to cast_assoc/3.